add onChange prop to Wrapper component, with tests

timweightman opened the pull request:
Related to #48 

Basic implementation of the suggestion in [this comment](https://github.com/davidtheclark/react-aria-menubutton/issues/48#issuecomment-249017391), as favored by [the owner](https://github.com/davidtheclark/react-aria-menubutton/issues/48#issuecomment-254020228)

- [x] update README.md
- [x] update failing test snapshots

timweightman commented:
@davidtheclark Please let me know if you there is anything I can do to help with this PR.

davidtheclark commented:
Sorry for the delay. I will try to take a closer look tomorrow morning.

davidtheclark commented on lib/createManager.js:
```diff
@@ -20,6 +20,7 @@ var protoManager = {
       externalStateControl.registerManager(this.options.id, this);
     }
 
+    this.onChange = options ? options.onChange : null;
```
This is unnecessary because `options` is already attached to `this` at line 13.

davidtheclark commented on README.md:
````diff
@@ -241,6 +241,30 @@ function handleSelection(value, event) {
 }
 ```
 
+**onChange** { Function }: A callback to run when the menu is opened or closed. It will be passed the `isOpen` state of the menu, e.g. `{ isOpen: true }`
````
I don't think `onChange` is a clear and specific name for this prop. `onMenuToggle` might be more clear.

Is there a reason to have one prop function that receives an object with state instead of two such functions: `onMenuOpen` and `onMenuClose`?

timweightman replied:
Happy to change to `onMenuToggle`.

Regarding `onMenuOpen` / `onMenuClose` that would have been overkill for our use case. It's easy enough for consumers either way. As an API, I like to have as few hooks as possible- easier to learn. This one hook can service both open / close, consumer can branch in their handler if needs be.
You have final say. Will respect if you prefer to split it up.

timweightman commented:
@davidtheclark I've updated according to your feedback. For now I have refactored `onChange` to `onMenuToggle` as advised, for the reasons in my previous comment.

If you would prefer to split it into `onMenuOpen` and `onMenuClose`, I should be able to manage that at some point in the coming week.

I'm not sure what caused CI to fail with my latest checkin, can you advise if this is something I need to action, and if so, how? 

🍻 🍻 🍻 cheers

davidtheclark commented:
@timweightman Looks good, thanks! I will try to merge, clean up the tests, and release later this weekend or early next week.

davidtheclark commented:
Published in 4.3.0.

timweightman commented:
Thanks @davidtheclark, you're a champ!

On Sun, 27 Nov 2016, 08:40 David Clark <notifications@github.com> wrote:

> Merged #56
> <https://github.com/davidtheclark/react-aria-menubutton/pull/56>.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/davidtheclark/react-aria-menubutton/pull/56#event-872699810>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGXqC3Dny8o74e41nB0yEQZXx1j94ZUPks5rCKdMgaJpZM4KyJmz>
> .
>
